src/main.js

```diff
@@ -272,9 +272,22 @@ const CodingManagerApplication = new Lang.Class({
                                      'org.freedesktop.DBus.Properties',
                                      'PropertiesChanged',
                                      propChangedVariant);
+        if (this.Visible) {
+            if (!this._changedSignalId) {
+                this._changedSignalId = Wnck.Screen.get_default().connect('active-window-changed', Lang.bind(this, this._on_active_window_changed));
```

Comment:
I wonder if we're always getting this signal as soon as the connection is made? Or if there is some event which triggers this signal immediately